[1.11.x] Fixed #28550 -- Restored contrib.auth's login() and logout() views' respect of positional arguments.

ZachLiuGIS opened the pull request:
Added old login() and logout() views signature arguments back and add
them to kwargs to be passed to LoginView() and LogoutView()
https://code.djangoproject.com/ticket/28550

timgraham commented:
I think something like this is on the right track: http://dpaste.com/14AVKK0. Additional tests are needed to say for sure.

ZachLiuGIS commented:
SimpleTestCase will cause this error, I used TestCase for TestLogin

`AssertionError: Database queries aren't allowed in SimpleTestCase. Either use TestCase or TransactionTestCase to ensure proper test isolation or set TestLogin.allow_database_queries to True to silence this failure.`

I also updated LogoutView and added TestLogout.

timgraham commented on django/contrib/auth/views.py:
```diff
@@ -133,12 +133,21 @@ def get_context_data(self, **kwargs):
+        redirect_field_name=redirect_field_name,
+        form_class=authentication_form,
+        extra_context=extra_context,
+        redirect_authenticated_user=redirect_authenticated_user
```
Include a trailing comma in cases like this so that if more items are added later, this line doesn't have to be modified again.

timgraham commented on django/contrib/auth/views.py:
```diff
@@ -133,12 +133,21 @@ def get_context_data(self, **kwargs):
         RemovedInDjango21Warning, stacklevel=2
     )
-    return LoginView.as_view(**kwargs)(request, *args, **kwargs)
+
```
Don't add a blank line.

timgraham commented on tests/auth_tests/test_deprecated_views.py:
```diff
@@ -4,16 +4,18 @@
 import datetime
 import itertools
 import re
+from importlib import import_module
```
Remove unused import.

ZachLiuGIS commented:
Cleaned up code based on comments.

Added more tests for login and logout. 

the attributes not in tests are (not sure how to test these):

- redirect_field_name (in login() and logout())
- redirect_authenticated_user (in login())

timgraham commented:
Thanks! Given that the views are deprecated and not receiving further changes, I think the tests you provided are sufficient.
